mir_eval/transcription.py

```diff
@@ -0,0 +1,376 @@
+'''
+The aim of a transcription algorithm is to produce a symbolic representation of
+a recorded piece of music in the form of a set of discrete notes. There are
+different ways to represent notes symbolically. Here we use the piano-roll
+convention, meaning each notes has a start time, a duration (or end time), and
+a single, constant, pitch value. Pitch values can be quantized (e.g. to a
+semitone grid tuned to 440 Hz), but do not have to be. Also, the transcription
+can contain the notes of a single instrument or voice (for example the melody),
+or the notes of all instruments/voices in the recording. This module is
+instrument agnostic: all notes in the estimate are compared against all notes
+in the reference.
+
+There are many metrics for evaluating transcription algorithms. Here we limit
+ourselves to the most simple and commonly used: given two sets of notes, we
+count how many estimate notes match the reference, and how many do not. Based
+on these counts we compute the precision, recall, and f-measure of the estimate
+given the reference. The default criteria for considering two notes to be a
+match are adopted from the `MIREX Multiple fundamental frequency estimation and
+tracking, Note Tracking subtask (task 2) <http://www.music-ir.org/mirex/wiki/\
+2015:Multiple_Fundamental_Frequency_Estimation_%26_Tracking_Results_-_MIREX_\
+Dataset#Task_2:Note_Tracking_.28NT.29>`_:
+
+"This subtask is evaluated in two different ways. In the first setup , a
+returned note is assumed correct if its onset is within +-50ms of a ref note
+and its F0 is within +- quarter tone of the corresponding reference note,
+ignoring the returned offset values. In the second setup, on top of the above
+requirements, a correct returned note is required to have an offset value
+within 20% of the ref notes duration around the ref note's offset, or within
+50ms whichever is larger."
+
+In short, we compute precision, recall and f-measure, once without taking
+offsets into account, and the second time with.
+
+For further details see Salamon, 2013 (page 186), and references therein:
+
+    Salamon, J. (2013). Melody Extraction from Polyphonic Music Signals.
+    Ph.D. thesis, Universitat Pompeu Fabra, Barcelona, Spain, 2013.
+
+Note: two different evaluation scripts have been used in MIREX over the years,
+where one uses ``<`` for matching onsets, offsets, and pitch values, whilst
+the other uses ``<=`` for these checks. That is, if the distance between two
+onsets is exactly equal to the defined threshold (e.g. 0.05), then the former
+script would not consider the two notes to have matching onsets, whilst the
+latter would. `mir_eval` provides both options: by default the latter
+(``<=``) is used, but you can set ``strict=True`` when calling
+``precision_recall_f1()`` in which case ``<`` will be used. The default value
```

Comment:
Make this a function link, like so:

    :func:`mir_eval.transcription.precision_recall_f1()`